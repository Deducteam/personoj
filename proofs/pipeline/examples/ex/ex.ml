let () =
  match
    Unix.system
      (Filename.quote_command "psnj-pipe"
         [ "--qfo"; "encoding/qfo.json"; "ex.log" ])
  with
  | Unix.WEXITED 0 -> 
      ignore @@ Unix.system (Filename.quote_command "cat" ["exi.lp"])
  | Unix.WEXITED n ->
      Format.eprintf "Command exited with code %d@." n;
      exit 1
  | Unix.WSIGNALED _ | Unix.WSTOPPED _ ->
      Format.eprintf "Command stopped by signal";
      exit 2
