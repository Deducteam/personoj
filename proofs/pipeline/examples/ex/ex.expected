symbol exi!1 : @Prf (@imp (imp qfo.spec.main.P qfo.spec.main.P) (@imp (disj (not qfo.spec.main.P) qfo.spec.main.P) (@imp (imp qfo.spec.main.P qfo.spec.main.P) (ex prop (λ (_ : El prop), imp qfo.spec.main.P qfo.spec.main.P))))) ≔
begin
  why3;
end;
symbol exi!2 : @Prf (imp qfo.spec.main.P qfo.spec.main.P) ≔
begin
  why3;
end;
symbol exi!3 : @Prf (@imp (imp qfo.spec.main.P qfo.spec.main.P) (@imp (disj (not qfo.spec.main.P) qfo.spec.main.P) (@imp (imp qfo.spec.main.P qfo.spec.main.P) (imp qfo.spec.main.P qfo.spec.main.P)))) ≔
begin
  why3;
end;
symbol exi!4 : @Prf (disj (not qfo.spec.main.P) qfo.spec.main.P) ≔
begin
  why3;
end;
symbol exi!5 : @Prf (imp qfo.spec.main.P qfo.spec.main.P) ≔
begin
  why3;
end;